### PR TITLE
Upgrade memdumps & controller

### PR DESCRIPTION
**Pull request by maciekpac**
### Types of changes

- [ ] Bugfix (non-breaking change which fixes issue linked in Description above)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [x] Cosmetics (whitespace, appearance)
- [ ] Other

### Checklist

- [ ] Code compiles without errors
- [ ] All tests pass locally with my changes (see TESTING section in CONTRIBUTING file)
- [ ] Created tests which will fail without the change (if possible)
- [ ] Extended the README/documentation (if necessary)
- [ ] All newly added files have proprietary license (if necessary)
- [ ] All newly added files are referenced in MANIFEST files (if necessary)

## Further comments

---
This change is [Reviewable](https://reviewable.io/reviews/bratpiorka/memkind_hot_poc/10)

### CODE REVIEW

**Review by maciekpac — COMMENTED** [review]
Master is broken, needs to be fixed before this pull request can be merged.

Most of the changes are bogus - they already existed on master before master was broken.

I would block my own pull request until master is fixed, but I can't :rofl:

File: include/memkind/internal/ranking_controller.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    double hotTierSize;
+    double coldTierSize;
+} ranking_controller;
+
+/// @p gain 1) has no effect (0;1) lowers response, (1;inf) amplifies response
+extern void ranking_controller_init_ranking_controller(
+    ranking_controller *controller, double expected_dram_total, double gain);
+
+extern double ranking_controller_calculate_fixed_thresh(
+    ranking_controller *info, double found_dram_total);
```
Comment (PatKamin) [inline include/memkind/internal/ranking_controller.h:22]: Rename `info` to `controller`.

File: include/memkind/internal/ranking_controller.h [hunk 2]
```diff
@@ -5,14 +5,18 @@ extern "C" {
...
+//     double derivative; // This would require filtering!!!
     double hotTierSize;
     double coldTierSize;
 } ranking_controller;
 
 /// @p gain 1) has no effect (0;1) lowers response, (1;inf) amplifies response
 extern void ranking_controller_init_ranking_controller(
-    ranking_controller *controller, double expected_dram_total, double gain);
+    ranking_controller *controller, double expected_dram_total,
+    double proportinal_term, double integral_term);
```
Comment (PatKamin) [inline include/memkind/internal/ranking_controller.h:19]: nit: typo in `proportinal_term`

File: src/ranking_controller.c [hunk 3]
```diff
@@ -6,8 +6,11 @@
 
 
 MEMKIND_EXPORT void ranking_controller_init_ranking_controller(
-    ranking_controller *controller, double expected_dram_total, double gain) {
-    controller->gain = gain;
+    ranking_controller *controller, double expected_dram_total,
+    double proportinal_term, double integral_term) {
```
Comment (PatKamin) [inline src/ranking_controller.c:10]: ditto

**Review by maciekpac — COMMENTED** [review]
Done

> *[Reviewable](https://reviewable.io/reviews/bratpiorka/memkind_hot_poc/10#-:-Mo8RC7I6h2MGk8qC6o2:bb6un1j)* status: 0 of 26 files reviewed, 2 unresolved discussions (waiting on @PatKamin)

___
*[include/memkind/internal/ranking_controller.h, line 18 at r2](https://reviewable.io/reviews/bratpiorka/memkind_hot_poc/10#-Mo4jwPZEGLrTpTBQb45-r2-18:-Mo8S2ZU3DXihRooMn-Y:b-896fix) ([raw file](https://github.com/bratpiorka/memkind_hot_poc/blob/e3bbca138bb8b5493bc4564eb100b0246617c7a2/include/memkind/internal/ranking_controller.h#L18)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

Rename `info` to `controller`.
</blockquote></details>

Done.
___
*[src/ranking_controller.c, line 10 at r1](https://reviewable.io/reviews/bratpiorka/memkind_hot_poc/10#-Mo4jwP_GYNyLvD3-1z9-r1-10:-Mo8Qdb74KXKbLOp0VTL:b-896fix) ([raw file](https://github.com/bratpiorka/memkind_hot_poc/blob/46de2901f137a0f6d9caca422c38fa7737fb05e1/src/ranking_controller.c#L10)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.

**Review by bratpiorka — APPROVED** [review]
> Reviewed 2 of 24 files at r1, 17 of 21 files at r2, 2 of 4 files at r3, 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bratpiorka/memkind_hot_poc/10)* status: all files reviewed, 2 unresolved discussions (waiting on @PatKamin)

**Review by PatKamin — APPROVED** [review]
> Reviewed 2 of 24 files at r1, 2 of 21 files at r2, 2 of 4 files at r3, 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bratpiorka/memkind_hot_poc/10)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @maciekpac)

**Review by PatKamin — COMMENTED** [review]
Rename this as well: https://github.com/bratpiorka/memkind_hot_poc/blob/3a7fe6e8779beb9b694110efd17aa86e78975825/include/memkind/internal/memkind_memtier.h#L292

**Review by maciekpac — COMMENTED** [review]
Done.

> *[Reviewable](https://reviewable.io/reviews/bratpiorka/memkind_hot_poc/10#-:-Mo8qtB30AwC06g_eBiD:b-896fix)* status: 22 of 26 files reviewed, all discussions resolved (waiting on @bratpiorka and @PatKamin)

**Review by bratpiorka — APPROVED** [review]
:lgtm:

> Reviewed 4 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bratpiorka/memkind_hot_poc/10#-:-Mo8v5dh4v2OvP7Xr3uF:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @maciekpac)

**Comment by kilobyte** [comment]
> I would block my own pull request until master is fixed, but I can't :rofl:

You can change it to a draft.

**Pull request merged**